src/osd/OSD.cc

```diff
@@ -7316,6 +7246,11 @@ void OSD::handle_pg_notify(OpRequestRef op)
       continue;
     }
 
+    uint64_t features;
+    if (from == get_nodeid())	// from myself?
+      features = CEPH_FEATURES_SUPPORTED_DEFAULT;
+    else
+      features = op->get_req()->get_connection()->get_features();
```

Comment:
I kept that the same from the existing code.  I hit one of the new asserts because an osd can notify itself, and in that case the features in the connection is 0.